modules/browser-node/src/channelProvider.ts

```diff
@@ -0,0 +1,280 @@
+import EventEmitter from "eventemitter3";
+import {
+  ChannelRpcMethod,
+  EngineEvent,
+  ChannelRpcMethods,
+  EngineEvents,
+  EngineEventMap,
+  IVectorEngine,
+  EngineParams,
+} from "@connext/vector-types";
+import { constructRpcRequest, safeJsonParse } from "@connext/vector-utils";
+
+export function isEventName(event: string): event is EngineEvent {
+  return event in EngineEvents;
+}
+
+export function isMethodName(event: string): event is ChannelRpcMethod {
+  return event in ChannelRpcMethods;
+}
+
+export interface IframeOptions {
+  id: string;
+  src: string;
+}
+
+export function payloadId(): number {
+  const date = new Date().getTime() * Math.pow(10, 3);
+  const extra = Math.floor(Math.random() * Math.pow(10, 3));
+  return date + extra;
+}
```

Comment:
Is this something we have to do in the server node as well?

Reply:
Nope this is only for the iframe <> parent communication through postMessage.